packages/components/src/components/vertical-tab/vertical-tab.jsx

```diff
@@ -9,101 +9,109 @@ import VerticalTabHeaderTitle from './vertical-tab-header-title.jsx';
 import VerticalTabLayout from './vertical-tab-layout.jsx';
 import VerticalTabWrapper from './vertical-tab-wrapper.jsx';
 
-class VerticalTab extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = { vertical_tab_index: props.vertical_tab_index || 0 };
+const setSelectedIndex = ({ current_path, list, is_routed, selected_index, setCurrTabIndex, setVerticalTabIndex }) => {
+    let index;
+    if (typeof selected_index === 'undefined') {
+        index = is_routed
+            ? Math.max(
+                  list.indexOf(list.find(item => item.path === current_path) || list.find(item => item.default)),
+                  0
+              )
+            : 0;
+    } else {
+        index = typeof selected_index === 'object' ? list.indexOf(selected_index) : selected_index;
     }
 
-    setSelectedIndex = ({ list, selected_index, is_routed, current_path }) => {
-        let index;
-        if (typeof selected_index === 'undefined') {
-            index = is_routed
-                ? Math.max(
-                      list.indexOf(list.find(item => item.path === current_path) || list.find(item => item.default)),
-                      0
-                  )
-                : 0;
-        } else {
-            index = typeof selected_index === 'object' ? list.indexOf(selected_index) : selected_index;
-        }
+    setCurrTabIndex(index);
 
-        this.setState({ vertical_tab_index: index });
+    if (typeof setVerticalTabIndex === 'function') {
+        setVerticalTabIndex(index);
+    }
+};
 
-        if (typeof this.props.setVerticalTabIndex === 'function') {
-            this.props.setVerticalTabIndex(index);
-        }
-    };
+const VerticalTab = ({
+    action_bar,
+    action_bar_classname,
+    current_path,
+    header_classname,
+    header_title,
+    is_floating,
+    is_full_width,
+    is_routed,
+    is_sidebar_enabled,
+    list,
+    list_groups,
+    setVerticalTabIndex,
+    tab_headers_note,
+    vertical_tab_index,
+}) => {
+    const [curr_tab_index, setCurrTabIndex] = React.useState(vertical_tab_index || 0);
+    const [selected_element, setSelectedElement] = React.useState(list[0]);
```

Comment:
we don't need an extra state for `selected_element`. 
we can just set `selected_element` from `list[curr_tab_index]`